src/model/docker.ts

```diff
@@ -95,7 +99,173 @@ class Docker {
         ${sshAgent ? '--volume /home/runner/.ssh/known_hosts:/root/.ssh/known_hosts:ro' : ''} \
         ${image}`;
 
-    await exec(command, undefined, { silent });
+        await exec(linuxRunCommand, undefined, { silent });
+        break;
+      }
+      case 'win32': {
+        let unitySerial = '';
+        if (!process.env.UNITY_SERIAL) {
+          //No serial was present so it is a personal license that we need to convert
+          if (!process.env.UNITY_LICENSE) {
+            throw new Error(`Missing Unity License File and no Serial was found. If this
+                             is a personal license, make sure to follow the activation
+                             steps and set the UNITY_LICENSE GitHub secret or enter a Unity
+                             serial number inside the UNITY_SERIAL GitHub secret.`);
+          }
+          unitySerial = this.getSerialFromLicenseFile(process.env.UNITY_LICENSE);
+        } else {
+          unitySerial = process.env.UNITY_SERIAL!;
+        }
+        core.setSecret(unitySerial);
+
+        if (!(process.env.UNITY_EMAIL && process.env.UNITY_PASSWORD)) {
+          throw new Error(`Unity email and password must be set for Windows based builds to
+                           authenticate the license. Make sure to set them inside UNITY_EMAIL
+                           and UNITY_PASSWORD in Github Secrets and pass them into the environment.`);
+        }
+
+        await this.setupWindowsRun(platform);
+
+        this.validateWindowsPlatformRequirements(platform);
+
+        const windowsRunCommand = `docker run \
+        --workdir c:/github/workspace \
+        --rm \
+        --env UNITY_LICENSE \
+        --env UNITY_LICENSE_FILE \
+        --env UNITY_EMAIL \
+        --env UNITY_PASSWORD \
+        --env UNITY_SERIAL="${unitySerial}" \
+        --env UNITY_VERSION="${version}" \
```

Comment:
Looks like we do in fact not need a switch (baseOs) here. The parameters should be passed taking the baseOs in mind.

That and other checks currently happen in the factory method (`create`) of the build-parameters model.

We should either make the OS-modifications to the parameters inside `build-parameters` or create another layer in between for OS specific stuff only. 

Personally I think 
- `unitySerial` can be parsed inside`build-parameters`
- `--workdir c:/github/workspace \` can be specific to the `docker` class, but like `getWorkDirParam(buildPlatform)` or something similar
- `await this.setupWindowsRun(platform);` can also be handled in `docker` class but like `setupBuildPlatform(buildPlatform)`

Note that these would not be functional changes, just moving around some code to make it more manageable.

Reply:
The big change on the windows branch is passing in all of the windows specific directories like Visual Studio, win10sdk registry keys, etc. How do we get that through without a branch?

Reply:
My suggestion would be to use a branch, but more localised. 

For example to a method that returns a string with the platform specific only.

```
docker.run(`
   same
   same
   same
   ${platformSpecificDirectories(baseOs)}
`)
```